src/x/instrument/config_prometheus.go

```diff
@@ -328,3 +337,21 @@ func (g *multiGatherer) Gather() ([]*dto.MetricFamily, error) {
 
 	return results, nil
 }
+
+func appendCommonLabelsForMetric(commonLabels map[string]string, metric *dto.Metric) {
+	for l := range commonLabels {
+		label := l
+		labelValue := commonLabels[label]
+		metric.Label = append(metric.Label, &dto.LabelPair{Name: &label, Value: &labelValue})
```

Comment:
not sure if its safe to mutate Metric without any locks. Any thoughts?

Reply:
It looks ok to me